web/docs/database/operations/actions.md

```diff
@@ -0,0 +1,889 @@
+  fn: import { markTaskAsDone } from "@server/actions.js"
+}
+
+```
+
+</TabItem>
+<TabItem value="ts" label="TypeScript">
+
+```wasp title="main.wasp"
+// ...
+
+action createTask {
+  fn: import { createTask } from "@server/actions.js"
+}
+
+action markTaskAsDone {
+  fn: import { markTaskAsDone } from "@server/actions.js"
+}
+```
+
+:::warning
+Even though you are using TypeScript and plan to implement this Action in `src/server/actions.ts`, you still need to import it using a `.js` extension. Wasp internally uses `esnext` module resolution, which requires importing all files with a `.js` extension. This is only needed when importing `@server` files.
+
+Read more about ES modules in TypeScript [here](https://www.typescriptlang.org/docs/handbook/esm-node.html). If you're interested in the discussion and the reasoning behind this, read about it [in this GitHub issue](https://github.com/microsoft/TypeScript/issues/33588).
+:::
+
+</TabItem>
+</Tabs>
+
+<small>
+
+If you want to know about all supported options for the `action` declaration, take a look at the [Options reference](#options-reference).
+
+</small>
+
+Wasp Actions and their implementations don't need to (but can) have the same name. We'll keep the names identical to avoid confusion.
+
+:::info
+You might have noticed that we told Wasp to import Action implementations that don't yet exist. Don't worry about that for now. We'll write the implementations imported from `actions.{js,ts}` in the next section.
+
+It's a good idea to start with the high-level concept (i.e., the Action declaration in the Wasp file) and only then deal with the implementation details (i.e., the Action's implementation in JavaScript).
+:::
+
+After declaring a Wasp Action, two crucial things happen:
+
+- Wasp **generates a server-side NodeJS function** that shares its name with the Action. This function's interface is identical to the client-side function from the previous point.
+- Wasp **generates a client-side JavaScript function** that shares its name with the Action (e.g., `createTask`).
+  This function takes a single optional argument - an object containing any serializable data you wish to use inside the Action.
+  Wasp will send this object over the network and pass it into the Action's implementation as its first positional argument (more on this when we look at the implementations).
+  Such an abstraction works thanks to an HTTP API route handler Wasp generates on the server, which calls the Action's NodeJS implementation under the hood.
+
+Generating these two functions ensures a uniform calling interface across the entire app (both client and server).
+
+### Implementing Actions in Node
+
+Now that we've declared the Action, all that's left is implementing it.
+We've told Wasp to look for Actions' implementations in the file `src/server/actions.{js,ts}`, so that's where we should export them from.
+
+Here's how you might implement the previously declared Actions `createTask` and `markTaskAsDone`:
```

Comment:
```suggestion
Now that we've declared the Action, what remains is to implement it. We've instructed Wasp to look for the Actions' implementations in the file `src/server/actions.{js,ts}`, so that's where we should export them from.

Here's how you might implement the previously declared Actions `createTask` and `markTaskAsDone`:
```